cmd/minishift/cmd/image/config_add.go

```diff
@@ -0,0 +1,66 @@
+/*
+Copyright (C) 2017 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package image
+
+import (
+	"github.com/spf13/cobra"
+
+	"fmt"
+	"github.com/minishift/minishift/cmd/minishift/cmd/config"
+	"github.com/minishift/minishift/pkg/util/os/atexit"
+	"github.com/minishift/minishift/pkg/util/strings"
+)
+
+var (
+	addConfiguredImageCmd = &cobra.Command{
+		Use:   "add [image ...]",
+		Short: "Adds the specified images to the list of configured images for import and export.",
+		Long:  "Adds the specified images to the list of configured images for import and export.",
+		Run:   addConfiguredImage,
+	}
+)
+
+func addConfiguredImage(cmd *cobra.Command, args []string) {
+	if len(args) == 0 {
+		atexit.ExitWithMessage(1, noImageSpecified)
+	}
+
+	configMap, err := config.ReadConfig()
+	if err != nil {
+		atexit.ExitWithMessage(1, fmt.Sprintf("Cannot read the Minishift configuration: %s", err.Error()))
+	}
+
+	var cacheImages []string
+	if configMap[config.CacheImages.Name] == nil {
+		cacheImages = []string{}
+	} else {
+		cacheImages = toStringSlice(configMap[config.CacheImages.Name].([]interface{}))
```

Comment:
This modifies the general config... right? Why no consolidate all this logic in the actual `config.go` as a utility function instead of having this logic here. This creates an additional entrypoint to the config file.

But also, why is it not part of `AllInstances`?

Reply:
> This modifies the general config... right? Why no consolidate all this logic in the actual config.go

Because it is handled by the image commands. It just uses the same file. Same for add-ons. 

> This creates an additional entrypoint to the config file.

Where is the issue? 

> But also, why is it not part of AllInstances?

Because this is a per instance configuration, not for all instances.

Reply:
We have an instance level config

Reply:
https://github.com/minishift/minishift/tree/master/pkg/minishift/config

Reply:
> Why no consolidate all this logic in the actual config.go as a utility function instead of having this logic here.

BTW, I am already using the  functions ReadConfig and WriteConfig from the config to read/write. So I am not re-inveting the wheel. That said, what the image command does with the config belongs into image.

Reply:
> Same for add-ons.

And this is perhaps wrong. The general config should actually contain 'general' config related to the VM and the setup of the VM. It can be used to enable functionality, but we have a config for instance and all instances specific settings. which contains config related to how we want Minishift to be used.

Reply:
> BTW, I am already using the functions ReadConfig and WriteConfig 

If you already use these functions, then why not consolidate:

I know you use:
   "github.com/minishift/minishift/cmd/minishift/cmd/config"

So my argument stands as is.

Unless you would use:
   "github.com/minishift/minishift/tree/master/pkg/minishift/config"

it would have made more sense. Since it doesn't use the actual config commands, I think you should not use these functions as currently implemented. And I would rather have seen you use the `minishiftConfig` for instances or allinstances.